src/include/buffer.h

```diff
@@ -50,6 +50,8 @@
 #include <exception>
 #include <type_traits>
 
+#include <boost/intrusive/list.hpp>
```

Comment:
this adds the dependency from `librados3-dev` to `libboost-dev`.  see http://pulpito.ceph.com/kchai-2018-11-25_10:39:02-rados-wip-kefu2-testing-2018-11-22-1635-distro-basic-smithi/3282147/

```
2018-11-25T11:15:17.574 INFO:tasks.workunit.client.0.smithi088.stdout:[100%] Building CXX object CMakeFiles/rocksdb-shared.dir/utilities/env_librados.cc.o
2018-11-25T11:15:17.672 INFO:tasks.workunit.client.0.smithi088.stderr:In file included from /usr/include/rados/librados.hpp:11:0,
2018-11-25T11:15:17.673 INFO:tasks.workunit.client.0.smithi088.stderr:                 from /home/ubuntu/cephtest/mnt.0/client.0/tmp/rocksdb/include/rocksdb/utilities/env_librados.h:14,
2018-11-25T11:15:17.673 INFO:tasks.workunit.client.0.smithi088.stderr:                 from /home/ubuntu/cephtest/mnt.0/client.0/tmp/rocksdb/utilities/env_librados.cc:4:
2018-11-25T11:15:17.673 INFO:tasks.workunit.client.0.smithi088.stderr:/usr/include/rados/buffer.h:53:36: fatal error: boost/intrusive/list.hpp: No such file or directory
2018-11-25T11:15:17.673 INFO:tasks.workunit.client.0.smithi088.stderr: #include <boost/intrusive/list.hpp>
2018-11-25T11:15:17.673 INFO:tasks.workunit.client.0.smithi088.stderr:                                    ^
2018-11-25T11:15:17.673 INFO:tasks.workunit.client.0.smithi088.stderr:compilation terminated.
```

so question is:

1. is it okay to have this dependency? @jdurgin @neha-ojha @liewegas  what do you think?
1. if yes, shall we depend on `libboost1.67-dev` or just `libboost-dev`? this implies following question. @rzarzynski could you help answer them?
   1. is `ptr_node` part of the API?
   1. is it okay to change its layout across librados versions?
   1. is `boost::intrusive::list_base_hook<>` an interface consistent across libboost1.65 and libboost1.67

Reply:
My answers:
1. yes, `ptr_node` becomes a part the API (because of `buffer::list::buffers_t` and the new variant of `bufferlist::push_back`).
2. I'm 100% sure here but I think the same rules would apply as to `buffer::ptr` and `buffer::list::iterator_impl<...>`. No difference between these entities in the matter, IMHO.
3. It is. Particular files are at the same commit IDs. Please see:
    * `boost/intrusive/list_hook.hpp` in [1.65](https://github.com/boostorg/intrusive/blob/boost-1.65.0/include/boost/intrusive/list_hook.hpp) and [1.67](https://github.com/boostorg/intrusive/blob/boost-1.67.0/include/boost/intrusive/list_hook.hpp),
    * `boost/intrusive/detail/generic_hook.hpp` in [1.65](https://github.com/boostorg/intrusive/blob/boost-1.65.0/include/boost/intrusive/detail/generic_hook.hpp) and [1.67](https://github.com/boostorg/intrusive/blob/boost-1.67.0/include/boost/intrusive/detail/generic_hook.hpp).

The alternative might be to use`std::list` with a custom allocator at the cost of flexibility or just dissect/freeze the `intrusive_list`.

Reply:
Rebased and dropped the dependency on the doubly linked `boost::intrusive::list`. It turned out `bufferlist` uses few methods of its interface and switching to a singly linked list was possible.